### PR TITLE
Calian updates to sdnotify

### PR DESCRIPTION
**Pull request by zack-mayoh**
- Added/updated copyright info
- Renamed to `async-sdnotify`
- Updated type annotations
- Converted to Poetry project
- Wrote unit tests for 100% test coverage:
```
============================= test session starts ==============================
platform linux -- Python 3.11.7, pytest-8.3.3, pluggy-1.5.0 -- /home/mayoh/async-sdnotify/.venv/bin/python
cachedir: .pytest_cache
rootdir: /home/mayoh/async-sdnotify
configfile: pyproject.toml
plugins: asyncio-0.24.0, cov-5.0.0
asyncio: mode=Mode.STRICT, default_loop_scope=None
collecting ... collected 5 items

tests/test_systemdnotifier.py::test_does_nothing_without_environment PASSED [ 20%]
tests/test_systemdnotifier.py::test_connect_throws_exception_when_debug PASSED [ 40%]
tests/test_systemdnotifier.py::test_status_throws_exception_when_debug PASSED [ 60%]
tests/test_systemdnotifier.py::test_start_watchdog_throws_exception_when_debug PASSED [ 80%]
tests/test_systemdnotifier.py::test_sends_notifications PASSED           [100%]

---------- coverage: platform linux, python 3.11.7-final-0 -----------
Name                      Stmts   Miss  Cover
---------------------------------------------
aiosdnotify/__init__.py      82      0   100%
---------------------------------------------
TOTAL                        82      0   100%
Coverage XML written to file coverage.xml


============================== 5 passed in 1.06s ===============================
Finished running tests!
```

### CODE REVIEW

File: aiosdnotify/__init__.py [hunk 1]
```diff
@@ -36,29 +46,31 @@ def __init__(self, debug=False, warn_limit=3, watchdog=True):
         self._warnings = 0
         self._transport: asyncio.DatagramTransport | None = None
         self._protocol: asyncio.DatagramProtocol | None = None
-        self._watchdog_task: asyncio.Task | None = None
+        self._watchdog_task: asyncio.Task[Any] | None = None
```
Comment (zack-mayoh) [inline aiosdnotify/__init__.py]: I could not figure out what the proper type parameter for the Task generic is here. `asyncio.create_task` returns `Task[_T@create_task]`, but that doesn't seem to work.

Comment (bwillenborg) [reply aiosdnotify/__init__.py]: I think it's just the return type of the function in the task. In this case I think it's just `None`?

Comment (zack-mayoh) [reply aiosdnotify/__init__.py]: That would make sense

**Review by bwillenborg — CHANGES_REQUESTED** [review]
Change `asyncio.Task[Any]` to `asyncio.Task[None]`.

File: aiosdnotify/__init__.py [hunk 2]
```diff
@@ -78,35 +92,40 @@ def _notify(self, state_str: str):
...
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> Self:
         await self.connect()
         if self.watchdog:
             self.start_watchdog()
         return self
 
-    async def __aexit__(self, exc_type, exc_val, exc_tb):
+    async def __aexit__(self, exc_type: Optional[Type[BaseException]], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]) -> None:
```
Comment (alex-calian) [inline aiosdnotify/__init__.py]: Minor nitpick - Suggest  `... | None` to stay consistent in notation

File: tests/test_systemdnotifier.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    os.environ['NOTIFY_SOCKET'] = '@watchdog'
+    os.environ['WATCHDOG_USEC'] = '1000000'
+    yield sock
+    sock.close()
+    del os.environ['NOTIFY_SOCKET']
+    del os.environ['WATCHDOG_USEC']
+
+
+@pytest.mark.asyncio
+async def test_does_nothing_without_environment() -> None:
```
Comment (alex-calian) [inline tests/test_systemdnotifier.py]: This test isn't explicitly testing anything?

File: aiosdnotify/__init__.py (hunk 1 above)
Comment (zack-mayoh) [reply aiosdnotify/__init__.py]: Fixed

File: aiosdnotify/__init__.py (hunk 2 above)
Comment (zack-mayoh) [reply aiosdnotify/__init__.py]: Fixed

File: tests/test_systemdnotifier.py (hunk 3 above)
Comment (zack-mayoh) [reply tests/test_systemdnotifier.py]: Tried to make it a little clearer what the intention of this test is. Does it make sense?

**Review by bwillenborg — APPROVED** [review]

**Review by alex-calian — APPROVED** [review]

**Pull request merged**